src/connectors/stats/__tests__/connectStats-test.js

```diff
@@ -33,6 +39,155 @@ See documentation: https://www.algolia.com/doc/api-reference/widgets/stats/js/#c
     });
   });
 
+  describe('getRenderState', () => {
+    test('returns the render state', () => {
+      const renderFn = jest.fn();
+      const unmountFn = jest.fn();
+      const createStats = connectStats(renderFn, unmountFn);
+      const stats = createStats();
+      const helper = jsHelper(createSearchClient(), 'indexName', {
+        index: 'indexName',
+      });
+      helper.search = jest.fn();
+
+      const renderState1 = stats.getRenderState(
+        { stats: {} },
+        createInitOptions({ helper })
+      );
+
+      expect(renderState1.stats).toEqual({
+        hitsPerPage: undefined,
+        nbHits: 0,
+        nbPages: 0,
+        page: 0,
+        processingTimeMS: -1,
+        query: '',
+        widgetParams: {},
+      });
+
+      stats.init(createInitOptions({ helper, state: helper.state }));
+
+      const results = new SearchResults(helper.state, [
+        createSingleSearchResponse(),
+      ]);
+
+      const renderState2 = stats.getRenderState(
+        { stats: {} },
+        createRenderOptions({
+          helper,
+          state: helper.state,
+          results,
+        })
+      );
+
+      expect(renderState2.stats).toEqual({
+        hitsPerPage: results.hitsPerPage,
```

Comment:
Here we're just asserting that we wrote the right code but not the right feature. We should use actual values instead of variables because we can't understand if the right values are passed here.

Reply:
It makes sense, as there's no queries/results I can just return default values.

Should it also be the case with results?

Reply:
Yes—I would create test results in the `createSingleSearchResponse()` call to make sure that the right values are returned from `getRenderState`.

Reply:
Updated in https://github.com/algolia/instantsearch.js/pull/4565/commits/ab81329b6e51822e82ea5e7a3b4a144329d7dbb3